### ISSUE TITLE
Help troubleshooting "Response signature validation failed"

### ISSUE THREAD

**Issue by douglasrw** [onelogin/ruby-saml#494]
I am trying to to implement SSO, and I get an error "Response signature validation failed.
" on the samlresponse.   I get this both on our server and using the validation tool at https://www.samltool.com/validate_response.php#.  The identity provider claims that the signature is good.  How can I troubleshoot this?

**Comment by pitbulk** [comment]
Were you able to review that the x509cert info on the SAMLResponse matches the one you used for validating the Signature?
Any idea about the software used at the IdP side?

**Comment by douglasrw** [comment]
Thank you for your response!  In fact, the x509cert in the response **did not** match the x509cert in the MetaData file they provided with!

For testing, I have updated the IdP MetaData xml to include the x509cert sent in the response.

When I plug this into the response validator(https://www.samltool.com/validate_response.php) it now validates: "THE SAML RESPONSE IS VALID"

  Unfortunately, testing on our server we get a slightly different error now: "Invalid Signature on SAML Response".  I have checked to ensure that the x509cert in the response is the same as what is loaded from the MetaData xml file, and they are the same.  Any ideas what could cuase this?  Thanks in advance!
